porch/controllers/packagevariantsets/api/v1alpha1/packagevariantset_types.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	kptfilev1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+	pkgvarapi "github.com/GoogleContainerTools/kpt/porch/controllers/packagevariants/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// PackageVariantSet represents an upstream package revision and a way to
+// target specific downstream repositories where a variant of the upstream
+// package should be created.
+type PackageVariantSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PackageVariantSetSpec   `json:"spec,omitempty"`
+	Status PackageVariantSetStatus `json:"status,omitempty"`
+}
+
+func (o *PackageVariantSet) GetSpec() *PackageVariantSetSpec {
+	if o == nil {
+		return nil
+	}
+	return &o.Spec
+}
+
+// PackageVariantSetSpec defines the desired state of PackageVariantSet
+type PackageVariantSetSpec struct {
+	Upstream *Upstream `json:"upstream,omitempty"`
+	Targets  []Target  `json:"targets,omitempty"`
+
+	AdoptionPolicy pkgvarapi.AdoptionPolicy `json:"adoptionPolicy,omitempty"`
+	DeletionPolicy pkgvarapi.DeletionPolicy `json:"deletionPolicy,omitempty"`
+}
+
+type Upstream struct {
+	Package *Package `json:"package,omitempty"`
+
+	Revision string `json:"revision,omitempty"`
+
+	Tag string `json:"ref,omitempty"`
+}
+
+type Target struct {
+	// option 1: an explicit repo/package name pair
+	Package *Package `json:"package,omitempty"`
+
+	// option 2: a label selector against a set of repositories
+	Repositories map[string]string `json:"repositories,omitempty"`
```

Comment:
If we want to use label selectors, we probably want to do it the same as in other kubernetes APIs: https://github.com/GoogleContainerTools/kpt/blob/main/porch/controllers/rootsyncrollouts/api/v1alpha1/rootsyncrollout_types.go#L29

Reply:
Done, thanks for the link!